hack/remote-debug/README.md

```diff
@@ -0,0 +1,487 @@
+# Debug Development Environment
+
+This directory provides a container-based development environment
+designed to streamline the workflow for debugging the
+`openshift-router` process. It primarily contains a Dockerfile and
+some helper scripts and utilities. The Dockerfile produces an image
+that allows you to rapidly redeploy the `openshift-router` binary
+without needing to build a new image each time you want to debug an
+edit/compile iteration.
+
+## Rationale
+
+Traditionally, iterating on edit/compile/debug is slow because it
+involves building a new image, pushing the image, and waiting for the
+roll out to complete. This new environment provides a one-time deploy
+of the debug image. Subsequent iterations only require editing and
+compiling as before. The debugging step just copies a new binary to
+the pod and starts executing immediately, eliminating the need to wait
+for container builds, tagging an image, pushing an image, setting a
+new image on the deployment, and waiting for new pods to roll out.
+
+## Overview
+
+The debug image runs `tini` as the init system. `tini` launches
+`remote-debug-helper`, which performs the following tasks:
+
+1. Discovers all environment variables associated with the
+   `router-default` deployment.
+
+2. Writes a file named `/etc/profile.d/router-default.sh` that contains all of
+   `router-default`'s environment variables, ensuring arbitrary SSH
+   login shells get an identical environment as with `oc rsh`.
+
+3. Starts `sshd` for SSH access into the pod for development and
+   debugging.
+
+This setup allows you to run the `openshift-router` binary without
+specifying command line flags or environment variables, ensuring a
+friction-free invocation that yields an identical environment as a
+non-debug-based Dockerfile.
+
+## Preparatory Steps
+
+Before you begin, there are several preparatory steps you need to
+follow to set up your development environment.
+
+### 1. Obtain the `dlv` Binary
+
+    The `dlv` (Delve debugger) executable needs to be located in the
+    top-level directory as it is copied into the image during the
+    container build process. Use the provided helper script to find and
+    copy the `dlv` binary:
+
+    ```sh
+    $ ./hack/remote-debug/find-and-copy-dlv
```

Comment:
GoLand may be not present on the laptop (e.g. my laptop). This file may need to highlight the fact that it works only for GoLand users.

Reply:
Noted in 55c2b49793b2011a458b81e812be9c6f4f671b33.